src/cdomains/valueDomain.ml

```diff
@@ -148,10 +148,10 @@ struct
     | TComp ({cstruct=true; _} as ci,_) -> `Struct (Structs.create (fun fd -> init_value fd.ftype) ci)
     | TComp ({cstruct=false; _},_) -> `Union (Unions.top ())
     | TArray (ai, None, _) ->
-      `Array (CArrays.make (IndexDomain.bot ())  (if get_bool "ana.base.partition-arrays.enabled" then (init_value ai) else (bot_value ai)))
+      `Array (CArrays.make (IndexDomain.bot ())  (if (get_string "exp.arrays-domain"="partitioned" || get_string "exp.arrays-domain"="unroll") then (init_value ai) else (bot_value ai)))
```

Comment:
@michael-schwarz This is starting look like a leaky abstraction. Couldn't the array domains implement it in their corresponding modules? Currently the array domain selection is repeated 5 times instead of 1.

Reply:
I agree that it is a bit messy. To avoid this, we would have to make the array domains not take a `Lattice.S` but a new module type that also exposes `init_value` and `top_value` (or just add it to the `LatticeWithSmartOps` and use that one everywhere), and then in turn include `init_value` and `top_value` again in the signature of the `ArrayDomain`.

I added a note in https://github.com/goblint/analyzer/issues/485 that we want to do this together with the general cleanup of the partitioned array domain.